metals/src/main/scala/scala/meta/internal/metals/PackageProvider.scala

```diff
@@ -272,43 +276,117 @@ class PackageProvider(
     new TextEdit(range, importString)
   }
 
+  private def calcAllImportsEditsForFileMove(
+      importersAndParts: Vector[(Importer, Vector[String])],
+      oldPkgParts: Vector[String],
+      refsNames: Set[String],
+      refsSymbols: Set[String],
+      newPkgName: String,
+      source: AbsolutePath,
+  ): ImportEdits = {
+    val directlyImportedNames =
+      importersAndParts
+        .withFilter(_._2 == oldPkgParts)
+        .flatMap(_._1.importees)
+        .collect {
+          case Importee.Name(name) => name
+          case Importee.Rename(from, _) => from
+          case Importee.Unimport(name) => name
```

Comment:
Actually, this means that `name` is not imported, so maybe we shouldn't include it?

Reply:
Since in case of a wildcard we import all elements explicitly instead of using an `_` or `*`, we should simply delete the unimport?

Reply:
If we manage to import everything correctly then yeah. Without wildcard, unimport doesn't make any sense.

Reply:
I changed that in the other place but I it's needed here. Those are used to filter out the symbols referenced in the file that did not have a direct import (were imported using a wildcard). Unimproved symbol will also generate an occurrence (I think), so we do want to filter it out.